diff --git a/node_modules/nitrogen/lib/syntax/kotlin/KotlinStruct.js b/node_modules/nitrogen/lib/syntax/kotlin/KotlinStruct.js
index 17cef22..4b443ab 100644
--- a/node_modules/nitrogen/lib/syntax/kotlin/KotlinStruct.js
+++ b/node_modules/nitrogen/lib/syntax/kotlin/KotlinStruct.js
@@ -15,7 +15,7 @@ export function createKotlinStruct(structType) {
         return `
 @DoNotStrip
 @Keep
-val ${name}: ${type.getTypeCode('kotlin', false)}
+val ${name}: ${type.getTypeCode('kotlin', false)} ${type.type.kind === 'optional' ? '= null ' : ''}
 `.trim();
     })
         .join(',\n');
diff --git a/node_modules/nitrogen/src/syntax/kotlin/KotlinStruct.ts b/node_modules/nitrogen/src/syntax/kotlin/KotlinStruct.ts
index 02616bf..c9cc000 100644
--- a/node_modules/nitrogen/src/syntax/kotlin/KotlinStruct.ts
+++ b/node_modules/nitrogen/src/syntax/kotlin/KotlinStruct.ts
@@ -24,7 +24,7 @@ export function createKotlinStruct(structType: StructType): SourceFile[] {
       return `
 @DoNotStrip
 @Keep
-val ${name}: ${type.getTypeCode('kotlin', false)}
+val ${name}: ${type.getTypeCode('kotlin', false)} ${type.type.kind === 'optional' ? '= null ' : ''}
 `.trim()
     })
     .join(',\n')
